pkg/sa/agent.go

```diff
@@ -0,0 +1,60 @@
+package sa
+
+import (
+	"context"
+
+	"github.com/rancher/system-agent/pkg/applyinator"
+	"github.com/rancher/system-agent/pkg/config"
+	"github.com/rancher/system-agent/pkg/image"
+	"github.com/rancher/system-agent/pkg/k8splan"
+	"github.com/rancher/system-agent/pkg/localplan"
+	"github.com/rancher/system-agent/pkg/version"
+	"github.com/sirupsen/logrus"
+)
+
+type Agent struct {
+	cfg *config.AgentConfig
+}
+
+func (a *Agent) Run(ctx context.Context) error {
+
+	if a.cfg == nil {
+		return nil
```

Comment:
Should we error out with a nil config instead? What's the real world scenario when this is ran with a nil config?

Reply:
If the agent config isn't set, aka nil, then it's a no op. There is a decision record for t design.